feature/experimental/qos/qos_ingress_traffic_classification/ate_tests/ingress_traffic_classification_test/ingress_traffic_classification_test.go

```diff
@@ -0,0 +1,1089 @@
+	frameSize   uint32
+	codePoint   uint8
+	queue       string
+}
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// Test cases:
+//
+// Verify the presence of the telemetry paths of the following features:
+//  -  Configure DUT with ingress and egress routed interfaces.
+//  -  Configure QOS classifiers to match packets arriving on DUT ingress interface to corresponding forwarding class according to
+//     classification table.
+//  -  Configure packet re-marking for configured classes according to the marking table.
+//  -  One-by-one send flows containing every TOS/TC/EXP value in the classification table.
+//  -  For every flow sent, verify match-packets counters on the DUT ingress interface
+//  -  Verify packet markings on ATE ingress interface.
+//  -  Verify that no traffic drops in all flows
+//  - Check the QoS queue counters exist and are updated correctly
+//    - /qos/interfaces/interface/output/queues/queue/state/transmit-pkts
+//    - /qos/interfaces/interface/output/queues/queue/state/transmit-octets
+//    - /qos/interfaces/interface/output/queues/queue/state/dropped-pkts
+//    - /qos/interfaces/interface/output/queues/queue/state/dropped-octets
+//    - /qos/interfaces/interface/input/classifiers/classifier/terms/term/state/matched-packets
+//    - /qos/interfaces/interface/input/classifiers/classifier/terms/term/state/matched-octets
+//
+// Topology:
+//   ate:port1 <--> port1:dut:port2 <--> ate:port2
+//
+//  Sample CLI command to get telemetry using gnmic:
+//   - gnmic -a ipaddr:50051 -u username -p password --skip-verify get \
+//      --path /components/component --format flat
+//   - gnmic tool info:
+//     - https://github.com/karimra/gnmic/blob/main/README.md
+//
+
+func TestQoSClassifier(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	dp1 := dut.Port(t, "port1")
+	dp2 := dut.Port(t, "port2")
+
+	// Configure DUT interfaces and QoS.
+	configureDUTIntf(t, dut)
+	configureQoS(t, dut)
+
+	// Configure ATE interfaces.
+	ate := ondatra.ATE(t, "ate")
+	ap1 := ate.Port(t, "port1")
+	ap2 := ate.Port(t, "port2")
+	top := ate.Topology().New()
+	intf1 := top.AddInterface("intf1").WithPort(ap1)
+	intf1.IPv4().
+		WithAddress("198.51.100.1/31").
+		WithDefaultGateway("198.51.100.0")
+	intf1.IPv6().
+		WithAddress("2001:db8::2/126").
+		WithDefaultGateway("2001:db8::1")
+	intf2 := top.AddInterface("intf2").WithPort(ap2)
+	intf2.IPv4().
+		WithAddress("198.51.100.3/31").
+		WithDefaultGateway("198.51.100.2")
+	intf2.IPv6().
+		WithAddress("2001:db8::6/126").
+		WithDefaultGateway("2001:db8::5")
+	top.Push(t).StartProtocols(t)
+
+	type qosVals struct {
+		be1, af1, af2, af3, af4, nc1 string
+	}
+
+	qos := qosVals{
+		be1: "BE1",
+		af1: "AF1",
+		af2: "AF2",
+		af3: "AF3",
+		af4: "AF4",
+		nc1: "NC1",
+	}
+
+	if dut.Vendor() == ondatra.JUNIPER {
+		qos = qosVals{
+			be1: "0",
+			af1: "4",
+			af2: "1",
+			af3: "5",
+			af4: "2",
+			nc1: "3",
+		}
+	}
+
+	IPv4TrafficFlow1 := map[string]*trafficData{
```

Comment:
local variables should start lowercase. You could rename them all to start lowercase, but actually might be best to just inline them into the cases -- they're not reused so not a strong case to declare them in advance.

Reply:
will update in the next commit

Reply:
updated from all variables such as "IPv4TrafficFlow1" to ipv4TrafficFlow1, these variables are being called from line 281, can you please point out how these can be inlined

Reply:
Sure, instead of

```
trafficFlows: ipv4TrafficFlow1,
```

You would have:

```
trafficFlows: map[string]*trafficData{
	"flow-ipv4-nc1": {frameSize: 1000, trafficRate: 1, codePoint: 6, queue: qos.nc1},
	"flow-ipv4-af4": {frameSize: 400, trafficRate: 4, codePoint: 4, queue: qos.af4},
	"flow-ipv4-af3": {frameSize: 300, trafficRate: 3, codePoint: 3, queue: qos.af3},
	"flow-ipv4-af2": {frameSize: 200, trafficRate: 2, codePoint: 2, queue: qos.af2},
	"flow-ipv4-af1": {frameSize: 1100, trafficRate: 1, codePoint: 1, queue: qos.af1},
	"flow-ipv4-be1": {frameSize: 1200, trafficRate: 1, codePoint: 0, queue: qos.be1},
},
```

And so on with the other variables.

Reply:
Need the string value to distinguish between ipv4 and ipv6 flows. line 348